improver/spotdata/build_spotdata_cube.py

```diff
@@ -45,9 +45,9 @@ def build_spotdata_cube(
     wmo_id,
     scalar_coords=None,
     neighbour_methods=None,
-    neighbour_methods_dim=1,
+    neighbour_methods_dim=0,
     grid_attributes=None,
-    grid_attributes_dim=2,
+    grid_attributes_dim=1,
```

Comment:
@bayliffe, do we require the flexibility to specify dimensions for these coords? Couldn't we instead just set the required order of coords (with some of them being optional) and get rid of the dim arguments?

Reply:
Sure, whatever works is fine by me. Add them in any old order, then use the enforce_coordinate_order function (as is currently done at the CLI level for some reason).